### PR TITLE
LPS-202176 | Test Automation | Master

### PR DESCRIPTION
**Pull request by andreefarias**
## Tickets: 
* [LPS-202175](https://liferay.atlassian.net/browse/LPS-202175) 
* [LPS-202176](https://liferay.atlassian.net/browse/LPS-202176) 
* [LPS-202735](https://liferay.atlassian.net/browse/LPS-202735) 

## Goal
Cover scenarios with Poshi tests related the new field called auto increment. 

Sreenshot

### CODE REVIEW

**Comment by andreefarias** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	tearDown {
+		ObjectAdmin.deleteAllCustomObjectsViaAPI();
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase]: You don't need to make a separate commit to create the testcase file, you can just do that while adding the first test

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase [hunk 2]
```diff
@@ -101,6 +101,99 @@ definition {
...
+		}
+
+		task ("and Given the auto increment field value and the ERC are modified in the JSON file") {
+			ImportExport.openImportExportAdmin();
+
+			ImportExport.gotoImport();
+
+			ImportExport.selectEntity(entityType = "C_Shoe (v1_0 - Liferay Object REST)");
+
+			UploadDependencyFile.uploadFile(fileName = "ImportedObjectEntriesShoes.json");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase:165]: Hey @andreefarias, we're trying to avoid using hardcoded object entries for our tests. The idea here is to modify the exported file and then import it.

**Review by timpak — CHANGES_REQUESTED** [review]
Hey @andreefarias can you redo `CanIncrementAfterImportingObjectEntry` so that it doesn't depend on a hardcoded file?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase (hunk 1 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase]: You don't need to make this change for this PR, just something to keep in mind moving forward.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase (hunk 2 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase:165]: See https://liferay.atlassian.net/wiki/spaces/QA/pages/2157085195/Common+Poshi+Methods#StringUtil to modify strings.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase (hunk 1 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase]: Ok, info saved 🧠.

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:stop

**Comment by andreefarias** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase [hunk 3]
```diff
@@ -101,6 +101,99 @@ definition {
...
+				value = "Blue");
+		}
+
+		task ("and Given the object entry is exported") {
+			ImportExport.exportAndUnzipFile(
+				entityType = "C_Shoe (v1_0 - Liferay Object REST)",
+				exportFileFormat = "JSON");
+		}
+
+		task ("and Given the auto increment field value and the ERC are modified in the JSON file") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase:149]: Hey @andreefarias, since we're changing the text field and not the ERC, can you update the wording here? Just replace `ERC` with `text field value`

**Review by timpak — CHANGES_REQUESTED** [review]
Hey @andreefarias one minor change since we updated the logic. Can you make this change and forward this?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase (hunk 3 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase:149]: Ok, I'll update.

Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldsAutoIncrement.testcase:149]: Wait... but I'm changing the ERC and not the text field.

ERC = `FileUtil.replaceStringInFile(${filePath}, "shoe001", "shoe020");`
Auto increment = `FileUtil.replaceStringInFile(${filePath}, "001-Sneakers", "020-Sneakers");`

**Comment by andreefarias** [comment]
ci:forward
